calm/dsl/cli/bps.py

```diff
@@ -326,27 +326,63 @@ def get_blueprint_runtime_editables(client, blueprint):
     return response.get("resources", [])
 
 
-def get_field_values(entity_dict, context, path=None):
+def get_field_values(entity_dict, context, path=None, input_runtime_editables=None):
     path = path or ""
     for field, value in entity_dict.items():
         if isinstance(value, dict):
             get_field_values(entity_dict[field], context, path=path + "." + field)
         else:
-            new_val = input(
-                "Value for {} in {} (default value={}): ".format(
-                    path + "." + field, context, value
-                )
+            new_val = get_val_input_runtime_editables(
+                input_runtime_editables, field, path, context
             )
+            if not new_val:
+                new_val = input(
+                    "Value for {} in {} (default value={}): ".format(
+                        path + "." + field, context, value
+                    )
+                )
             if new_val:
                 entity_dict[field] = type(value)(new_val)
 
 
+def get_val_input_runtime_editables(input_runtime_editables, field, path, context):
+    filtered_input_runtime_editables = list(
```

Comment:
So here we will not use original context. i.e.

As name of entity will be independent (service_name != peofile_name != package_name ....etc.)
So here context will be taken how we treat them in DSL.
Ex:
`runtime_vars = [
    {
        "value": {
                "value": "1"
            },
            "context": "DefaultProfile",
            "name": "replica_count"
    },
]`

At extend we have to expand it to _app_profile.DefaultProfile.variable_

Reply:
Basically context start will be the major entities in DSL i.e. services, packages, substrates etc. For every variable context u have to traverse from **PROFILE-> DEPLOYMENTS->....**,  to create exact context and use it for filling runtime_ediatables.